contracts/PangolinExchange.sol

```diff
@@ -4,15 +4,18 @@ pragma solidity ^0.8.2;
 import "@pangolindex/exchange-contracts/contracts/pangolin-periphery/interfaces/IPangolinRouter.sol";
 import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
 import "@openzeppelin/contracts/access/Ownable.sol";
+import "./IAssetsExchange.sol";
 
 /**
  * @title PangolinExchange
  * @dev Contract allows user to invest into an ERC20 token
  * This implementation uses the Pangolin DEX
  */
-contract PangolinExchange is Ownable {
+contract PangolinExchange is Ownable, IAssetsExchange {
   /* ========= STATE VARIABLES ========= */
   IPangolinRouter pangolinRouter;
+  mapping(bytes32 => address) assetToAddressMapping;
+  mapping(bytes32 => uint256) assetToDecimalPlacesMapping;
```

Comment:
"Decimals" is one of the property of ERC20 standard. Have you considered reading that from token contract?

Reply:
Actually the standard defines the decimals method as optional and mentiones`"OPTIONAL - This method can be used to improve usability, but interfaces and other contracts MUST NOT expect these values to be present.`.
Because of this it is available in the open zeppelin IERC20Detailed interface and not in the base one.

I would expect most of the real world tokens to use it, however in theory it is possible that we may want to include a token into our application but it won't have this method.

Reply:
As discussed, I implemented the usage of optional ERC20 .decimals() method